.changeset/silver-lies-rest.md

```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/plugin-catalog-backend-module-gitlab': minor
+---
+
+The configuration key `branch` of the `GitlabDiscoveryEntityProvider` has been deprecated in favor of the configuration key `fallbackBranch`.
+To migrate to the new configuration value, rename `branch` to `fallbackBranch`.
```

Comment:
It might be good to add a line stating that this config key will be used for X in a future release so that people get some additional context early on.